Source/WebCore/Modules/filesystemaccess/FileSystemWritableFileStreamSink.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "WritableStreamSink.h"
+
+namespace WebCore {
+class FileSystemFileHandle;
+
+class FileSystemWritableFileStreamSink final : public WritableStreamSink {
+public:
+    static ExceptionOr<Ref<FileSystemWritableFileStreamSink>> create(FileSystemFileHandle&);
+    ~FileSystemWritableFileStreamSink();
+
+private:
+    explicit FileSystemWritableFileStreamSink(FileSystemFileHandle&);
+    Ref<FileSystemFileHandle> protectedSource() const { return m_source; }
+
+    // WritableStreamSink
+    void write(ScriptExecutionContext&, JSC::JSValue, DOMPromiseDeferred<void>&&) final;
+    void close() final;
+    void error(String&&) final;
+
+    Ref<FileSystemFileHandle> m_source;
```

Comment:
Let's make it a `const Ref`, and then we might not need to use `protectedSource()` at all.

Reply:
It doesn't hurt to do it but I doubt Ryosuke has had time to implement this in the static analyzer yet.